.circleci/config.yml

```diff
@@ -91,6 +100,47 @@ jobs:
       - ms-teams/report:
           only_on_fail: true
           webhook_url: $MS_TEAMS_WEBHOOK_URL
+  build_prerelease_1:
+    working_directory: ~/openbas
+    docker:
+      - image: cimg/base:stable-20.04
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run: sudo apt-get update -qq && sudo apt install curl gettext-base
+      - run:
+          working_directory: ~/openbas
+          name: Replace pyobas requirement of stable version with latest release branch code
+          command: find . -name requirements.txt -exec sed 's|^pyobas==.*$|pyobas @ git+https://github.com/OpenBAS-Platform/client-python@${CIRCLE_BRANCH}|' -i {} \;
```

Comment:
the use of single quoting for the `sed` expression will insert the literal string `pyobas @ git+https://github.com/OpenBAS-Platform/client-python@${CIRCLE_BRANCH}`: this leaves pip to expand the env var by itself.

issue: the env vars are not passed to the docker build process, rather the available env vars in a docker build are those set in the base image. An env var may be set in the dockerfile with the `ENV` statement, but the actual value (that of the parent process' `$CIRCLE_BRANCH`) should be passed as a `--build-arg` (with a corresponding `ARG` statement in the Dockerfile)

```dockerfile
ARG branch_name
ENV CIRCLE_BRANCH $branch_name
```
```shell
docker build --build-arg "branch_name=${CIRCLE_BRANCH}" .
```


You may be tempted to use double quoting to have the shell expand `$CIRCLE_BRANCH` before replacing the matched text in files, hence
potential security issue: I would be able to create a branch called `release/something\nmaliciousdep @ evilrepo` and inject a malicious dependency at build time.